aea/decision_maker/base.py

```diff
@@ -110,42 +111,45 @@ def check_transaction_is_consistent(self, tx_message: TransactionMessage) -> boo
         """
         Check if the transaction is consistent.
 
-        E.g. check that the agent state has enough money if it is a buyer
-        or enough holdings if it is a seller.
+        E.g. check that the agent state has enough money if it is a buyer or enough holdings if it is a seller.
+        Note, the agent is the sender of the transaction message by design.
         :return: True if the transaction is legal wrt the current state, false otherwise.
         """
-        currency_id = tx_message.currency_id
-        if tx_message.is_sender_buyer:
-            # check if we have the money to cover amount and tx fee.
-            result = self.amount_by_currency[currency_id] >= tx_message.amount + tx_message.sender_tx_fee
-        else:
-            # check if we have the goods.
-            result = True
-            for good_id, quantity in tx_message.quantities_by_good_id.items():
-                result = result and (self.quantities_by_good_id[good_id] >= quantity)
-            # check if we have the money to cover tx fee.
-            result = self.amount_by_currency[currency_id] + tx_message.amount >= tx_message.sender_tx_fee
+        result = len(tx_message.tx_amount_by_currency_id) == 1
+        for currency_id, amount in tx_message.tx_amount_by_currency_id.items():
+            if amount <= 0 and all(quantity >= 0 for quantity in tx_message.tx_quantities_by_good_id.values()):
+                # check if the agent has the money to cover amount and tx fee (the agent is the buyer).
+                transfer_amount = -amount - tx_message.tx_counterparty_fee
+                result = result and (transfer_amount >= 0)
+                max_tx_fee = tx_message.tx_sender_fee + tx_message.tx_counterparty_fee
+                max_payable = transfer_amount + max_tx_fee
+                result = result and (self.amount_by_currency_id[currency_id] >= max_payable)
+            elif amount >= 0 and all(quantity <= 0 for quantity in tx_message.tx_quantities_by_good_id.values()):
```

Comment:
I noticed that both in the buyer and in the seller cases, the comparisons are `<=` and `>=` (equality included), respectively.
What happens if `amount = 0` and for all the quantities,  `quantity = 0`?

Reply:
Good point, I changed this now to reject the transaction for the special, all `0` case.